### PR TITLE
[extractor/tiktok] Fix and improve metadata extraction

### PR DESCRIPTION
**Pull request by bashonly**
* Fixes metadata extraction from video webpages (`_parse_aweme_video_web`):
     * `creator` was failing to be extracted (addresses [#6483 (comment)](https://github.com/yt-dlp/yt-dlp/issues/6483#issuecomment-1500164016))
     * `thumbnails` were failing to be extracted
     * `id` extraction was not fatal, and the value always matched from the URL could easily be passed as an argument to `_parse_aweme_video_web`

 * Fixes some potentially fatal metadata extraction from mobile API responses (`_parse_aweme_video_app`):
     * `thumbnails` extraction KeyErrored if a Douyin API response was received
     * `author_info = aweme_detail.get('author', {})` should be `aweme_detail.get('author') or {}` since `author` can be `None`, etc

 * Extracts the secondary user ID as `channel_id` from both web and app, which will be useful/necessary for a `TikTokUserIE` fix (it already would be useful for #4996)
 
 * Refactors and attempts to improve readability
 
 * Adds a webpage-only test case, updates tests
 
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/tiktok.py [hunk 1]
```diff
@@ -344,76 +349,76 @@ def extract_addr(addr, add_meta={}):
...
         formats = []
-        play_url = video_info.get('playAddr')
-        width = video_info.get('width')
-        height = video_info.get('height')
-        if isinstance(play_url, str):
-            formats = [{
+        width = int_or_none(video_info.get('width'))
+        height = int_or_none(video_info.get('height'))
+
+        for play_url in traverse_obj(video_info, ('playAddr', (({list}, ..., 'src'), ({str})), {url_or_none})):
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:364]: `({str})` -  the brackets are useless. Also, it's not needed due to `url_or_none` later
`{list}, ...` - what's the `{list}` for?

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/tiktok.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/tiktok.py:364]: `{list}` and `{str}` are used to branch depending on type, as `playAddr` can be a list or a string

Comment (pukkandan) [reply yt_dlp/extractor/tiktok.py:364]:
```suggestion
        for play_url in traverse_obj(video_info, ('playAddr', ((..., 'src'), None), {url_or_none})):
```

Is this what u want?

Comment (bashonly) [reply yt_dlp/extractor/tiktok.py:364]: yep, ig i overthought that one

**Pull request merged**